### PR TITLE
Homework10: Pavel Mitrafanau

### PR DESCRIPTION
**Pull request by Pavel2812mi**
Goode day, Sergei
Please review my decorators homework code

### CODE REVIEW

File: Homework10.py
Comment (Stml89) [inline Homework10.py:1]: Please, use a separate file for each task

File: Homework10.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+@validate_arguments
+def numbers(*args):
+    """
+    Функция, передающая аргументы декоратору для проверки
+    """
+    raise ValueError("Введен неверный аргумент")
+
+
+numbers(-5)
```
Comment (Stml89) [inline Homework10.py]: Verify that your code works via `assert`

File: Homework10.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+            print(f"Ошибка: {e}")
+    return wrapper
+
+
+@validate_arguments
+def numbers(*args):
+    """
+    Функция, передающая аргументы декоратору для проверки
+    """
+    raise ValueError("Введен неверный аргумент")
```
Comment (Stml89) [inline Homework10.py]: print arguments here

File: Homework10.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+        и переданные функции для выполнения операций
+    """
+    def typed(func):
+        def wrapper(*args):
+            converted_args = []
+            for arg in args:
+                if isinstance(arg, list):
+                    for item in arg:
+                        try:
+                            converted_args.append(expected_type(item))
```
Comment (Stml89) [inline Homework10.py]: nice, but you can use `map()` function to apply `int`, `str` to each element

**Review by Stml89 — CHANGES_REQUESTED** [review]

**Comment by Pavel2812mi** [comment]
Done, added 4 separate files for each task instead of one
Please review again

File: Homework10_2.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    Args:
+        Результат выполнения функцией операции над аргументами
+
+    Returns:
+        Возбуждает исключение ValueError
+        в случае, если результат функции не является числом
+        Возвращает функцию, если результат является числом
+    """
+    def wrapper(arg1, arg2):
```
Comment (Stml89) [inline Homework10_2.py]: better to expect `*args` different amount of arguments

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: Homework10_2.py (hunk 4 above)
Comment (Pavel2812mi) [reply Homework10_2.py]: Done

**Review by Stml89 — APPROVED** [review]

**Pull request merged**